force-app/main/releaseGating/classes/ReleaseGateAPIService.cls

```diff
@@ -105,16 +104,31 @@ public virtual with sharing class ReleaseGateAPIService {
      * @return           A GetReleaseGateActionResultModel
      */
     public virtual GetReleaseGateActionResultModel getReleaseGates() {
-        Object result = this.makeCall(
-            GET_ACTION,
-            new Map<String, Object>{ VERSION_PARAMETER => this.releaseGateRegistry.apiVersion }
-        );
-        GetReleaseGateActionResultModel resultModel = (GetReleaseGateActionResultModel) JSON.deserialize(
-            (String) result,
-            GetReleaseGateActionResultModel.class
-        );
-        this.verifyCallableResult(resultModel, GET_ACTION);
-        return resultModel;
+        //Customize application perm is required for all release gating actions
+        this.locateUserPermissionService().checkCustomizeApplicationForCurrentUser();
+        try {
+            Object result = this.makeCall(
+                GET_ACTION,
+                new Map<String, Object>{ VERSION_PARAMETER => this.releaseGateRegistry.apiVersion }
+            );
+            GetReleaseGateActionResultModel resultModel = (GetReleaseGateActionResultModel) JSON.deserialize(
+                (String) result,
+                GetReleaseGateActionResultModel.class
+            );
+            return resultModel;
+        } catch (System.JSONException je) {
+            String errorMessage = String.format(
+                Label.stgErrorSettingsProductInvalidResponse,
+                new List<Object>{ this.releaseGateRegistry.name }
+            );
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, je.getMessage());
+            throw new ReleaseGateCallableException(errorMessage, je, this.releaseGateRegistry, null);
+        } catch (ReleaseGateException rge) {
+            throw rge;
+        } catch (Exception ex) {
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, ex.getMessage());
```

Comment:
I think we could remove this debugging line

Reply:
Again here with a value added for CCE.

Reply:
oh great! good to know, thanks for explaining it